### PR TITLE
Manage multiple network messengers on network components

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- handle multiple network messengers 
  
## Proposed changes
- integrated latest mx-chain-communication-go and updated networkComponents to create and manage both messengers
- added new fullArchiveP2P.toml file(for the moment just a copy of p2p.toml) + new flags
- removed MetricP2PFullHistoryObservers which won't be needed anymore
- removed FullHistoryList from diffPeerListCreator as it won't be needed anymore added the needed disabled components
- added todos for further implementation

## Testing procedure
- will be tested with the feat branch

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: factory/network/networkComponents.go [hunk 1]
```diff
@@ -215,27 +191,33 @@ func (ncf *networkComponentsFactory) Create() (*networkComponents, error) {
...
+
+	mainNetworkComp.netMessenger.WaitForConnections(ncf.bootstrapWaitTime, ncf.mainP2PConfig.Node.MinNumPeersToWaitForOnBootstrap)
+
+	err = fullArchiveNetworkComp.netMessenger.Bootstrap()
 	if err != nil {
 		return nil, err
 	}
 
-	netMessenger.WaitForConnections(ncf.bootstrapWaitTime, ncf.p2pConfig.Node.MinNumPeersToWaitForOnBootstrap)
+	fullArchiveNetworkComp.netMessenger.WaitForConnections(ncf.bootstrapWaitTime, ncf.fullArchiveP2PConfig.Node.MinNumPeersToWaitForOnBootstrap)
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/network/networkComponents.go:206]: removed

File: factory/network/networkComponents.go [hunk 2]
```diff
@@ -253,6 +235,87 @@ func (ncf *networkComponentsFactory) createPeerHonestyHandler(
...
+		return networkComponentsHolder{}, err
+	}
+
+	argsMessenger := p2pFactory.ArgsNetworkMessenger{
+		ListenAddress:         ncf.listenAddress,
+		Marshaller:            ncf.marshalizer,
+		P2pConfig:             p2pConfig,
+		SyncTimer:             ncf.syncer,
+		PreferredPeersHolder:  peersHolder,
+		NodeOperationMode:     ncf.nodeOperationMode,
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/network/networkComponents.go:270]: right, removed it: https://github.com/multiversx/mx-chain-communication-go/pull/27

**Review by iulianpascalau — APPROVED** [review]

File: factory/network/networkComponents.go [hunk 3]
```diff
@@ -253,6 +234,86 @@ func (ncf *networkComponentsFactory) createPeerHonestyHandler(
 	return peerHonesty.NewP2pPeerHonesty(ratingConfig.PeerHonesty, pkTimeCache, suCache)
 }
 
+func (ncf *networkComponentsFactory) createNetworkHolder(
+	peersHolder p2p.PreferredPeersHolderHandler,
+	p2pConfig p2pConfig.P2PConfig,
+	logger p2p.Logger,
+) (networkComponentsHolder, error) {
+
```
Comment (ssd04) [inline factory/network/networkComponents.go:242]: remove this empty line?

File: factory/network/networkComponents.go [hunk 4]
```diff
@@ -215,27 +192,31 @@ func (ncf *networkComponentsFactory) Create() (*networkComponents, error) {
...
-	err = netMessenger.Bootstrap()
+	err = mainNetworkComp.netMessenger.Bootstrap()
 	if err != nil {
 		return nil, err
 	}
 
-	netMessenger.WaitForConnections(ncf.bootstrapWaitTime, ncf.p2pConfig.Node.MinNumPeersToWaitForOnBootstrap)
+	mainNetworkComp.netMessenger.WaitForConnections(ncf.bootstrapWaitTime, ncf.mainP2PConfig.Node.MinNumPeersToWaitForOnBootstrap)
+
+	err = fullArchiveNetworkComp.netMessenger.Bootstrap()
```
Comment (ssd04) [inline factory/network/networkComponents.go:202]: do we need to run WaitForConnections also for full achive net messenger?

File: factory/network/networkComponents.go (hunk 3 above)
Comment (sstanculeanu) [reply factory/network/networkComponents.go:242]: it is more readable this way, with the params split on multiple lines

File: cmd/node/config/fullArchiveP2P.toml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    #Port is the port that will be opened by the node on all interfaces so other peers can connect to it
+    #If the port = 0, the node will search for a free port on the machine and use it
+    Port = "37373-38383"
+
+    #ThresholdMinConnectedPeers represents the minimum number of connections a node should have before it can start
+    #the sync and consensus mechanisms
+    ThresholdMinConnectedPeers = 3
+
+    # MinNumPeersToWaitForOnBootstrap is the minimum number of peers to wait on bootstrap or the node will wait the default
+    # time which is now set to ~20 seconds (the const defined in the common package named TimeToWaitForP2PBootstrap)
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply cmd/node/config/fullArchiveP2P.toml]: pushed

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**